test/e2e/shoot/logging/event_logging.go

```diff
@@ -0,0 +1,207 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package logging
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/retry"
+	"github.com/gardener/gardener/test/e2e"
+	"github.com/gardener/gardener/test/framework"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	eventLogger            = "event-logger"
+	eventMessageForSeed    = "Event from Event Logger integration test related to the Seed"
+	eventMessageForShoot   = "Event from Event Logger integration test related to the Shoot"
+	seedEventName          = "seed-event"
+	shootEventName         = "shoot-event"
+	eventLoggerName        = "event-logger"
+	kubeSystemNamespace    = "kube-system"
+	operatorID             = "operator"
+	lokiName               = "loki"
+	gardenNamesapce        = "garden"
+	fluentBitDaemonSetName = "fluent-bit"
+)
+
+var _ = Describe("Seed logging testing", Label("Shoot", "logging"), func() {
+
+	var (
+		shootPurpose   = gardencorev1beta1.ShootPurposeDevelopment
+		shootNamespace string
+		lokiLabels     = map[string]string{
+			"app":  "loki",
+			"role": "logging",
+		}
+	)
+
+	f := defaultShootCreationFramework()
+	f.Shoot = e2e.DefaultShoot("event-")
+	f.Shoot.Spec.Purpose = &shootPurpose
+
+	BeforeEach(func() {
+		parentCtx = context.Background()
+
+		ctx, cancel := context.WithTimeout(parentCtx, 15*time.Minute)
+		defer cancel()
+		Expect(f.CreateShootAndWaitForCreation(ctx, false)).To(Succeed())
+		f.Verify()
+
+		shootNamespace = f.Shoot.Status.TechnicalID
+
+		Eventually(func(g Gomega) {
+			g.Expect(checkRequiredEventLoggerResources(ctx, f.ShootFramework.SeedClient, shootNamespace)).To(Succeed())
+		}).Should(Succeed())
+	})
+
+	It("should get events from loki", Label("event-logger"), func() {
+		ctx, cancel := context.WithTimeout(parentCtx, 15*time.Minute)
+		defer cancel()
+
+		By("Wait until Loki StatefulSet is ready")
+		Eventually(func(g Gomega) {
+			g.Expect(f.WaitUntilStatefulSetIsRunning(ctx, lokiName, shootNamespace, f.ShootFramework.SeedClient)).To(Succeed())
+		}).Should(Succeed())
```

Comment:
`WaitUntilStatefulSetIsRunning` already retries until the condition is met, so wrapping it in `Eventually` doesn't make much sense.
You could drop `Eventually` in this case.